### DISCUSSION TITLE
Nevermind

### DISCUSSION THREAD

**Discussion by Oddness-1** [godotengine/godot-proposals#10742] [Engine Core]

**Comment by Oddness-1** [comment]
Just a thumbs down? No pointed criticism of the idea @Jesusemora?

**Comment by AThousandShips** [comment]
This seems like just a lot of code to do some standard code behaviour and with little to no concrete idea for why it's *necessary* as opposed to it being *convenient*, the suggested cases are very simple to do with code and would be *more* complicated to implement with some complicated button class with "standard" behaviour that you need to configure

**Reply by Oddness-1** [reply]
Fair enough. Thanks for the explanation. It's not necessary, so much as convenient, like you said.

My implementation aside, the value of something *like* this is that, even if the user knows very little about code at all, you have the mechanisms already in place to implement an (admittedly simple) form of visual scripting, where anyone can make 'pretty pictures move, change color, or rotate on the screen,' while never writing a single line of code.

The suggestion was about accessibility, rather than what's necessary or convenient.

**Reply by Oddness-1** [reply]
P.S.
As far as there being a lot of code, quite a bit of that code was about working around certain limitations (Messing with owner, Multiplayer, etc. were causing errors, and I had to work around that)

**Reply by AThousandShips** [reply]
> even if the user knows very little about code at all

Designing for people who don't know how to code is a recipe for bloat and bad code, IMO you should *never* design like that when you make an engine, few will use it and few will maintain it

> while never writing a single line of code

I strongly disagree with that this is ever a goal

**Reply by Oddness-1** [reply]
Fair enough. Thanks. I'll mark the issue as closed.  :)
